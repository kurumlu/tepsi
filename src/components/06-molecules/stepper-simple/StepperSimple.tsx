import * as React from 'react';
import { FunctionComponent, ReactElement } from 'react';
import cn from 'classnames';

export type StepperSimpleProps = {
  activeStep?: number;
  dataQA?: string;
  size: number;
};

export const StepperSimple: FunctionComponent<StepperSimpleProps> = ({
  activeStep = 1,
  dataQA,
  size,
}) => {
  const listOfSteps: ReactElement[] = [];

  for (let i = 1; i < size + 1; ++i) {
    listOfSteps.push(
      <div
        className={cn('sd-m-stepper-simple__item', {
          'sd-m-stepper-simple__item--active': activeStep === i,
        })}
        key={i}
      />
    );
  }

  return (
    <div className="sd-m-stepper-simple" data-qa={dataQA}>
      {listOfSteps}
    </div>
  );
};
