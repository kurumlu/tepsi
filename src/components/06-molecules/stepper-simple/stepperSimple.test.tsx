import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { StepperSimple, StepperSimpleProps } from './StepperSimple';

describe('<StepperSimple />', () => {
  const component: ShallowWrapper<StepperSimpleProps> = shallow(
    <StepperSimple size={4} />
  );

  it('renders <StepperSimple /> component', () => {
    expect(component).toMatchSnapshot('default');
  });

  it('renders 4 stepper items', () => {
    expect(component.find('.as-m-stepper-simple__item')).toHaveLength(4);
  });

  it('adds active className for the first item if no prop "activeStep" provided', () => {
    expect(
      component.find('.as-m-stepper-simple__item').first().prop('className')
    ).toMatch(/as-m-stepper-simple__item--active/i);
  });

  it('adds active className for the last item if "activeStep" equals size', () => {
    const componentWithLastActiveStep: ShallowWrapper<StepperSimpleProps> = shallow(
      <StepperSimple size={4} activeStep={4} />
    );

    expect(
      componentWithLastActiveStep
        .find('.as-m-stepper-simple__item')
        .last()
        .prop('className')
    ).toMatch(/as-m-stepper-simple__item--active/i);
  });
});
