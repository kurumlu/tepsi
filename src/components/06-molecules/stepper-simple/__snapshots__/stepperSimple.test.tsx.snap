// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepperSimple /> renders <StepperSimple /> component: default 1`] = `
<div
  className="sd-m-stepper-simple"
>
  <div
    className="sd-m-stepper-simple__item sd-m-stepper-simple__item--active"
    key="1"
  />
  <div
    className="sd-m-stepper-simple__item"
    key="2"
  />
  <div
    className="sd-m-stepper-simple__item"
    key="3"
  />
  <div
    className="sd-m-stepper-simple__item"
    key="4"
  />
</div>
`;
